src/occupational_classification/lookup/soc_lookup.py

```diff
@@ -0,0 +1,287 @@
+"""This module provides the `SOCLookup` and `SOCRephraseLookup` classes, which facilitate
+the lookup of Standard Occupational Classification (SOC) codes based on descriptions and
+rephrased descriptions. It also handles preprocessing of SOC data and provides metadata
+for the classifications.
+
+Classes:
+    SOCLookup: A class for loading SOC data, performing lookups, and managing metadata.
+    SOCRephraseLookup: A class for performing rephrased lookups of SOC codes.
+
+"""
+
+from typing import Any, Optional, Union
+
+import pandas as pd
+
+from occupational_classification.data_access.soc_data_access import (
+    load_soc_index,
+    load_soc_structure,
+)
+from occupational_classification.hierarchy.soc_hierarchy import load_hierarchy
+from occupational_classification.meta.soc_meta import SocDB, SocMeta
+
+UNIT_CODE_LEN = 4
+
+
+class SOCLookup:
+    """A class for performing lookups of SOC codes based on descriptions.
+
+    Attributes:
+        data (pd.DataFrame): The SOC data loaded from a CSV file.
+        lookup_dict (dict[str, str]): A dictionary mapping descriptions to SOC codes.
+        meta (SocDB): Metadata for SOC classifications.
+
+    Methods:
+        lookup(description: str, similarity: bool = False) -> dict[str, Any]:
+            Looks up an SOC code based on the given description.
+    """
+
+    def __init__(
+        self,
+        data_path: str = "../src/occupational_classification/data/soc2020volume1structureanddescriptionofunitgroupsexcel16042025.xlsx",
+    ):
+        """Initialises the SOCLookup class by loading SOC data from a CSV file.
+
+        Args:
+            data_path (str): The path to the CSV file containing SOC data.
+        """
+        # Load data and store descriptions in lowercase
+        self.soc_df: pd.DataFrame = load_soc_structure(data_path)
+        self.data = self.data_preparation()
+        self.lookup_dict: dict[str, str] = self.data.set_index("description").to_dict()[
+            "label"
+        ]
+        self.meta: SocMeta = SocMeta(self.soc_df)
+
+    def data_preparation(self) -> pd.DataFrame:
```

Comment:
It seems to me that we should use soc_index as opposed to soc_structure. See example_sic_lookup_data.csv in industrial_classification.data, it's a lot like index in showing various known business activities mapped to a SIC code (non-unique).